### ISSUE TITLE
Error running example.

### ISSUE THREAD

**Issue by pandzel** [lwjglgamedev/lwjglbook#6]
I need a little help here. Examples build fine, however if running first example, it stops with the following error below. Neither natives.version set to 0.0.6 nor 0.0.7 helps. Any ideas?

Hello LWJGL 3.0.0 build 87!
[LWJGL] Version: 3.0.0 build 87
[LWJGL] 	 OS: Windows 10 v10.0
[LWJGL] 	JRE: 1.8.0_92 amd64
[LWJGL] 	JVM: Java HotSpot(TM) 64-Bit Server VM v25.92-b14 by Oracle Corporation
[LWJGL] Loading library (system): lwjgl
[LWJGL] 	Loaded from java.library.path: target\natives\lwjgl.dll
[LWJGL] Loading library: glfw
[LWJGL] ThreadLocalUtil state: UnsafeState
[LWJGL] MemoryUtil accessor: MemoryAccessorUnsafe
[LWJGL] 	Using SharedLibraryLoader...
[LWJGL] 	Extracting: file:/C:/Users/Piotr/.m2/repository/org/lwjgl/lwjgl-platform/3.0.0-SNAPSHOT/lwjgl-platform-3.0.0-SNAPSHOT-natives-windows.jar!/glfw.dll
java.lang.UnsatisfiedLinkError: Failed to load library: target\natives\glfw.dll (error code = 126)
	at org.lwjgl.system.windows.WindowsLibrary.<init>(WindowsLibrary.java:36)
	at org.lwjgl.system.APIUtil.apiCreateLibrary(APIUtil.java:110)
	at org.lwjgl.system.Library.loadNative(Library.java:187)
	at org.lwjgl.system.Library.loadNativeRelative(Library.java:163)
	at org.lwjgl.system.Library.loadNative(Library.java:141)
	at org.lwjgl.glfw.GLFW.<clinit>(GLFW.java:562)
	at org.lwjglb.game.Main.init(Main.java:44)
	at org.lwjglb.game.Main.run(Main.java:25)
	at org.lwjglb.game.Main.main(Main.java:115)
[LWJGL] Failed to load a library. Possible solutions:
	a) Set -Djava.library.path or -Dorg.lwjgl.librarypath to the directory that contains the shared libraries.
	b) Add the JAR(s) containing the shared libraries to the classpath.
Exception in thread "main" java.lang.NoClassDefFoundError: Could not initialize class org.lwjgl.glfw.GLFW
	at org.lwjglb.game.Main.run(Main.java:33)
	at org.lwjglb.game.Main.main(Main.java:115)

**Comment by lwjglgamedev** [comment]
Ok, please specify the OS and the JVM version. I guess that you're using Windows.

You should allways specificy where your natives are when runing the samples. In my case, since I use Netbeans, I've included a file named nbactions.xml in each chapter's source code that specifies where to look for them.

As you can see, it points to this directory "target\natives", and it's set by using the "-Djava.library.path" JVM argument. Allways use this. Another approach would be to get those binary libraries and set the PATH variable to include the idrectory where they are located.

I assume that you're using maven, if so, under "the target\natives" you should see the following files: glfw.dll, lwjgl.dll... etc. 

The thing is that, bye the stack trace you ar eposting, it seems that it's finding the libs but it cannot load them. A cause for the problema could be that yo are compiling using a 32 bits jvama and tty to execute with a 64 bits jvm version (or viceversa). Could you check this?

I'm using 32 bits JVMs, due to the requirements of other projects, and in the natives folder I see the 32 bits version for GLFW (glfw32.dll), LWJGL (lwjgl32.dll) and so on....

**Comment by pandzel** [comment]
I am using Windows 10 64bit. Also, JVM is 1.8.0_9 64 bits. "target/natives" contains both 64 and 32 versions of dll's. From the stack trace I am reading that it loaded lwjgl.dll but failed to load glfw.dll

**Comment by pandzel** [comment]
Well, I installed Java 32 bit version, compiled with it and trying run with it. The same issue (except log shows 32 versions of dll's being loaded.)

**Comment by lwjglgamedev** [comment]
I will test it in an environment similar to yours. In any case it seems strange to me that you have both 32 and 64 bits version in the natives folder. Are you sure that you are using correct JVM version for running and compiling. I've seen strange behaviours with old JRE installations messing around.

What IDE are you using?

**Comment by pandzel** [comment]
I am using NetBeans 8.1. I am confident about my JVM setup. To me it looks like glfw.dll can't be loaded; it's as simple as that. I don't think it is Java issue at all; it is .dll/OS issue. Perhaps glfw.dll has some dependencies it can not be resolved. So the real question is what exactly are these dependencies?

**Comment by lwjglgamedev** [comment]
Undesrtood, but please understand that I need to ask.

I dont' know which runtime dependencies GLFW depends on, but here are some things you could try:
- You could use Dependency Walker (http://www.dependencywalker.com/) to open the GLFW dll and check what's missing.
- You could try to install, if it's not the latest Microsoft Visual C++ Redesitributable package for both 32 and 64 bits. (I have this installed). I've found in the past problems  with libs that relly on this package.

Let me know if this helps.

**Comment by pandzel** [comment]
Dependencywalker showed a lot of missing dependencies. I installed VC++ Redistributable 2015 (all other former versions I've already got installed), but this didn't solved the issue. With so many missing dll's I thought it must be something different.

So, instead using "target/natives" I pointed java.library.path to use dll's which come from the official (stable) distribution of lwjgl. No issues here. I followed the same pattern for some other projects and it seems to be working.

**Comment by TheOnlySilverClaw** [comment]
Sounds like your configuration for LWJGL3 is wrong and I suggest you ask in that repository instead of here. ;)
But usually it should be enough to point the classpath to the natives.jar.

**Comment by lwjglgamedev** [comment]
A little late, but I have currenttly updated my equipment to Windows 10 64 bits and have _suffered_ the same issue as you.

It was not possible to load GLFW dll. I've tried several things and finally removed the need to use maven natives to copy native libraries. LWJGL already does this for us, DLLs are copied in a temporay directaroy at run time and loaded in that location.

I still don't know why it fails in this environment, but somehow DLLs get corrupted while using maven natives.

I've just updated the whole source code base and it should work in Windows 10 64 bits.
